packages/ui/src/memberships/modals/BuyMembershipModal/BuyMembershipFormModal.tsx

```diff
@@ -177,19 +177,53 @@ export const BuyMembershipForm = ({
     }
   }, [data?.membershipsConnection.totalCount])
 
-  const isFormValid = !isUploading && form.formState.isValid && (!isValidator || validatorAccounts?.length)
+  const isFormValid =
+    !isUploading &&
+    form.formState.isValid &&
+    (!isValidator ||
+      (validatorAccountCandidates?.filter((account) => !account || !isValidValidatorAccount(account)).length === 0 &&
+        validatorAccountCandidates?.filter((account) => !!account).length))
   const isDisabled =
     type === 'onBoarding' && process.env.REACT_APP_CAPTCHA_SITE_KEY ? !captchaToken || !isFormValid : !isFormValid
 
   const addValidatorAccount = () => {
-    if (validatorAccountCandidate && isValidValidatorAccount) {
-      setValidatorAccounts([...new Set([...validatorAccounts, validatorAccountCandidate])])
-      form?.setValue('validatorAccountCandidate' as keyof MemberFormFields, undefined)
-    }
+    setValidatorAccounts([...(validatorAccountCandidates ?? []), undefined])
+    form?.setValue('validatorAccountCandidate' as keyof MemberFormFields, undefined)
   }
 
   const removeValidatorAccount = (index: number) => {
-    setValidatorAccounts([...validatorAccounts.slice(0, index), ...validatorAccounts.slice(index + 1)])
+    validatorAccountCandidates &&
+      setValidatorAccounts([
+        ...validatorAccountCandidates.slice(0, index),
+        ...validatorAccountCandidates.slice(index + 1),
+      ])
+  }
+
+  const setValidatorAccounts = (accounts: (Account | undefined)[]) => {
+    form?.setValue('validatorAccountCandidates' as keyof MemberFormFields, [])
+    accounts.map((account, index) => {
+      form?.register(('validatorAccountCandidates[' + index + ']') as keyof MemberFormFields)
+      form?.setValue(('validatorAccountCandidates[' + index + ']') as keyof MemberFormFields, account)
+    })
+  }
+
+  const validatorAccountSelectorFilter = (index: number, account: Account) =>
+    (!validatorAccountCandidates ||
+      ![...validatorAccountCandidates.slice(0, index), ...validatorAccountCandidates.slice(index + 1)].find(
+        (accountOrUndefined) => accountOrUndefined?.address === account.address
+      )) &&
+    !!validatorAddresses?.includes(account.address)
+
+  const submit = () => {
+    const validatorAccounts = (validatorAccountCandidates?.filter(
+      (account, index, self) => !!account && index === self.findIndex((t) => t?.address === account.address)
+    ) ?? []) as Account[]
+    validatorAccounts.forEach((account, index) => {
+      form?.register(('validatorAccounts[' + index + ']') as keyof MemberFormFields)
+      form?.setValue(('validatorAccounts[' + index + ']') as keyof MemberFormFields, account)
+    })
```

Comment:
Then I think this should work:
```suggestion
    form?.setValue('validatorAccounts', accounts)
```